packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Watch,
+  Prop,
+  State,
+  h,
+} from '@stencil/core';
+import { assignLanguage, observerConfig } from '../../utils/utils';
+import { inheritAttributes } from '../../utils/utils';
+import i18n from './i18n/i18n';
+
+@Component({
+  tag: 'gcds-link',
+  styleUrl: 'gcds-link.css',
+  shadow: true,
+})
+export class GcdsLink {
+  @Element() el: HTMLElement;
+
+  private shadowElement?: HTMLElement;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Set the link size
+   */
+  @Prop({ mutable: true }) size: 'regular' | 'small' | 'inherit' = 'regular';
+
+  @Watch('size')
+  validateSize(newValue: string) {
+    const values = ['regular', 'small', 'inherit'];
+
+    if (!values.includes(newValue)) {
+      this.size = 'regular';
+    }
+  }
+
+  @Watch('display')
+  validateDisplay(newValue: string) {
+    const values = ['block', 'inline', 'inline-block'];
+
+    if (!values.includes(newValue)) {
+      this.display = 'block';
+    }
+  }
+
+  // @Watch('target')
+  // validateTarget(newValue: string) {
+  //   const values = ['_blank', '_self', '_parent', '_top'];
+  //
+  //   // TODO: if value is a valid string?
+  //   if (!values.includes(newValue)) {
+  //     this.target = '_self';
+  //   }
+  // }
+
+  /**
+   * Link props
+   */
+
+  /**
+   * Sets the display behavior of the link
+   */
+  @Prop() display: 'block' | 'inline' | 'inline-block' = 'inline';
+
+  /**
+   * The href attribute specifies the URL of the page the link goes to
+   */
+  @Prop() href!: string;
+
+  /**
+   * The rel attribute specifies the relationship between the current document and the linked document
+   */
+  @Prop() rel: string | undefined;
+
+  /**
+   * The target attribute specifies where to open the linked document
+   */
+  @Prop() target: string | undefined;
+
+  /**
+   * Whether the link is external or not
+   */
+  @Prop() external: false;
+
+  /**
+   * The download attribute specifies that the target (the file specified in the href attribute) will be downloaded when a user clicks on the hyperlink
+   */
+  @Prop() download: string | undefined;
+
+  /**
+   * The type specifies the media type of the linked document
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * Set additional HTML attributes not available in component properties
+   */
+  @State() inheritedAttributes: Object = {};
+
+  /**
+   * Language of rendered component
+   */
+  @State() lang: string;
+
+  /**
+   * Events
+   */
+
+  // TODO: Clarify all the necessary events
+
+  /**
+   * Emitted when the link has focus.
+   */
+  @Event() gcdsFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the link loses focus.
+   */
+  @Event() gcdsBlur!: EventEmitter<void>;
```

Comment:
@ethanWallace can I remove these?

Reply:
It would be best if the `gcds-link` emits those events on focus and on blur and even emit a click event as well. 

You would need something like the code below on the `a` tag. Might not be 100% correct, might have to play around with it.

``` jsx
<a
  onBlur={this.gcdsBlur.emit()}
  onFocus={this.gcdsFocus.emit()}
  onClick={this.gcdsClick.emit()}
```

Reply:
I'm looking around and found this, I'll try it out, thanks Ethan!
https://github.com/cds-snc/gcds-components/blob/develop/packages/web/src/components/gcds-search/gcds-search.tsx#L137-L139